src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUser_WithLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidLoginLength_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
```

Comment:
The test `registerUser_WithInvalidLoginLength_NotOk` does not align with the checklist as it does not mention a login length requirement. If there is a specific requirement for login length, it should be tested; otherwise, this test may not be necessary.